### DISCUSSION TITLE
Clicking "SSO" link leads to "Error. Check permissions." response rom Jellyfin

### DISCUSSION THREAD

**Discussion by ZacharyACoon** [9p4/jellyfin-plugin-sso#145] [Q&A] [answered]
Not sure what it's referring to, I expect it to create a new user based on roles?

**Comment by 9p4** [comment]
Please upload your logs. It shows what it is missing there.

**Reply by ZacharyACoon** [reply]
Jellyfin.Plugin.SSO_Auth.Api.SSOController: OpenID user "07d919b6-77af-4974-88c1-aeeec9009360" has one or more incorrect role claims: [{ Type: "jti", Value: "7fe9fa98-ddb5-4cc4-8db5-d02a36928f8f" }, { Type: "sub", Value: "07d919b6-77af-4974-88c1-aeeec9009360" }, { Type: "typ", Value: "ID" }, { Type: "azp", Value: "media.example.com" }, { Type: "session_state", Value: "4debebf6-41d8-4989-ac2e-d5d92ce390e8" }, { Type: "acr", Value: "0" }, { Type: "sid", Value: "4debebf6-41d8-4989-ac2e-d5d92ce390e8" }, { Type: "email_verified", Value: "True" }, { Type: "name", Value: "John Doe" }, { Type: "preferred_username", Value: "JohnDoe@gmail.com" }, { Type: "given_name", Value: "John" }, { Type: "family_name", Value: "Doe" }, { Type: "email", Value: "JohnDoe@gmail.com" }, { Type: "email_verified", Value: "true" }]. Expected any one of: ["media-watch"]

**Reply by 9p4** [reply]
Can you also upload your configuration? And also format them properly with backticks so it's easier to read?

**Reply by ZacharyACoon** [reply]
Jellyfin.Plugin.SSO_Auth.Api.SSOController: OpenID user "07d919b6-77af-4974-88c1-aeeec9009360" has one or more incorrect role claims: 
[{ Type: "jti", Value: "7fe9fa98-ddb5-4cc4-8db5-d02a36928f8f" },
 Type: "sub", Value: "07d919b6-77af-4974-88c1-aeeec9009360" },
 Type: "typ", Value: "ID" },
 Type: "azp", Value: "media.example.com" },
 Type: "session_state", Value: "4debebf6-41d8-4989-ac2e-d5d92ce390e8" },
 Type: "acr", Value: "0" },
 Type: "sid", Value: "4debebf6-41d8-4989-ac2e-d5d92ce390e8" },
 Type: "email_verified", Value: "True" },
 Type: "name", Value: "John Doe" },
 Type: "preferred_username", Value: "JohnDoe@gmail.com" },
 Type: "given_name", Value: "John" },
 Type: "family_name", Value: "Doe" },
 Type: "email", Value: "JohnDoe@gmail.com" },
 Type: "email_verified", Value: "true" }]. 

Expected any one of: ["media-watch"]

**Reply by 9p4** [reply]
What OIDC provider are you using?

**Reply by ZacharyACoon** [reply]
Keycloak

**Reply by 9p4** [reply]
Did you add the role mapper in keycloak?

**Reply by ZacharyACoon** [reply]
I see what the mapper is for, but I'm not sure how to configure it...  testing...

**Reply by ZacharyACoon** [reply]
I can't figure out how to configure this in keycloak....  I've been going in circles..

**Reply by ZacharyACoon** [reply]
I figured out a solution.   I might suggest readme updates.  Thank you.

**Reply by 9p4** [reply, marked as answer]
From providers.md:

> For adding groups and RBAC, go to the "mappers" tab, press "Add Builtin", and select either "Groups", "Realm Roles", or "Client Roles", depending on the role system you are planning on using. Once the mapper is added, edit the mapper and ensure that you note down the Token Claim Name as well as enable all four toggles: "Multivalued", "Add to ID token", "Add to access token", and "Add to userinfo" are enabled.

**Comment by ZacharyACoon** [comment]
These instructions are unclear in keycloak.
I might suggest Readme changes. :)
